packages/react-core/src/components/DatePicker/DatePicker.tsx

```diff
@@ -43,169 +44,190 @@ export interface DatePickerProps
   inputProps?: TextInputProps;
 }
 
+export interface DatePickerRef {
+  setCalendarOpen: (isOpen: boolean) => void;
+  toggleCalendar: () => void;
+}
+
 export const yyyyMMddFormat = (date: Date) =>
   `${date.getFullYear()}-${(date.getMonth() + 1).toString().padStart(2, '0')}-${date
     .getDate()
     .toString()
     .padStart(2, '0')}`;
 
-export const DatePicker: React.FunctionComponent<DatePickerProps> = ({
-  className,
-  locale = undefined,
-  dateFormat = yyyyMMddFormat,
-  dateParse = (val: string) => val.split('-').length === 3 && new Date(`${val}T00:00:00`),
-  isDisabled = false,
-  placeholder = 'YYYY-MM-DD',
-  value: valueProp = '',
-  'aria-label': ariaLabel = 'Date picker',
-  buttonAriaLabel = 'Toggle date picker',
-  onChange = (): any => undefined,
-  invalidFormatText = 'Invalid date',
-  helperText,
-  appendTo,
-  popoverProps,
-  monthFormat,
-  weekdayFormat,
-  longWeekdayFormat,
-  dayFormat,
-  weekStart,
-  validators = [],
-  rangeStart,
-  style: styleProps = {},
-  inputProps = {},
-  ...props
-}: DatePickerProps) => {
-  const [value, setValue] = React.useState(valueProp);
-  const [valueDate, setValueDate] = React.useState(dateParse(value));
-  const [errorText, setErrorText] = React.useState('');
-  const [popoverOpen, setPopoverOpen] = React.useState(false);
-  const [selectOpen, setSelectOpen] = React.useState(false);
-  const [pristine, setPristine] = React.useState(true);
-  const widthChars = React.useMemo(() => Math.max(dateFormat(new Date()).length, placeholder.length), [dateFormat]);
-  const style = { '--pf-c-date-picker__input--c-form-control--width-chars': widthChars, ...styleProps };
-  const buttonRef = React.useRef<HTMLButtonElement>();
-
-  React.useEffect(() => {
-    setValue(valueProp);
-    setValueDate(dateParse(valueProp));
-  }, [valueProp]);
-
-  const setError = (date: Date) => setErrorText(validators.map(validator => validator(date)).join('\n') || '');
+export const DatePicker = React.forwardRef<DatePickerRef, DatePickerProps>(
+  (
+    {
+      className,
+      locale = undefined,
+      dateFormat = yyyyMMddFormat,
+      dateParse = (val: string) => val.split('-').length === 3 && new Date(`${val}T00:00:00`),
+      isDisabled = false,
+      placeholder = 'YYYY-MM-DD',
+      value: valueProp = '',
+      'aria-label': ariaLabel = 'Date picker',
+      buttonAriaLabel = 'Toggle date picker',
+      onChange = (): any => undefined,
+      invalidFormatText = 'Invalid date',
+      helperText,
+      appendTo,
+      popoverProps,
+      monthFormat,
+      weekdayFormat,
+      longWeekdayFormat,
+      dayFormat,
+      weekStart,
+      validators = [],
+      rangeStart,
+      style: styleProps = {},
+      inputProps = {},
+      ...props
+    }: DatePickerProps,
+    ref
```

Comment:
I like this solution - for ease of future maintenance, could you format the forward ref in a pattern consistent with our other components? For example, [here is how we do it for Button](https://github.com/patternfly/patternfly-react/blob/fe2d3b55de1b60cf0c5740fb5180beece366d68e/packages/react-core/src/components/Button/Button.tsx#L163).

Reply:
Yes, I'll update it, thanks!

Reply:
I did split the declaration of the forward ref. Let me know if that's what you had in mind :)